src/draw.c

```diff
@@ -511,32 +511,65 @@ static void render_content(cairo_t *c, struct colored_layout *cl, int width)
         int h_text;
         pango_layout_get_pixel_size(cl->l, NULL, &h_text);
 
-        if (cl->icon && settings.icon_position == ICON_LEFT) {
-                cairo_move_to(c, cairo_image_surface_get_width(cl->icon) + 2 * settings.h_padding,
-                                 settings.padding + h/2 - h_text/2);
-        } else if (cl->icon && settings.icon_position == ICON_RIGHT) {
-                cairo_move_to(c, settings.h_padding, settings.padding + h/2 - h_text/2);
-        } else {
-                cairo_move_to(c, settings.h_padding, settings.padding);
+        int text_x = settings.h_padding,
+            text_y = settings.padding + h / 2 - h_text / 2;
+
+        // text positioning
+        if (cl->icon) {
+                // vertical alignment
+                switch (settings.content_alignment) {
+                        case CONTENT_TOP:
+                                text_y = settings.padding;
+                                break;
+                        case CONTENT_BOTTOM:
+                                text_y = h + settings.padding - h_text;
+                                if (text_y < 0) text_y = settings.padding;
```

Comment:
[nitpick] Please add a newline after if statements, it's hard to see the assignment there.

Reply:
I kinda expected this. ^^ That was also not meant to make it into the final version originally.